source/shared/core_results.cpp

```diff
@@ -81,78 +89,231 @@ SQLPOINTER read_lob_field( sqlsrv_stmt* stmt, SQLUSMALLINT field_index, sqlsrv_b
-                            sqlsrv_error_auto_ptr& last_error )
+SQLRETURN number_to_string( Number* number_data, _Out_ void* buffer, SQLLEN buffer_length, _Out_ SQLLEN* out_buffer_length, sqlsrv_error_auto_ptr& last_error )
 {
-	// get to display size by removing the null terminator from buffer length
-	size_t display_size = ( buffer_length - sizeof( Char )) / sizeof( Char );
+    size_t precision = 0;
 
+#ifdef _WIN32
     std::basic_ostringstream<Char> os;
-	// use the display size to determine the sql type. And if it is a double, set the precision accordingly
-	// the display sizes are set by the ODBC driver based on the precision of the sql type
-	// otherwise we can just use the default precision as long will not be truncated
-	size_t real_display_size = 14;
-	size_t float_display_size = 24;
-	size_t real_precision = 7;
-	size_t float_precision = 15;
-	// this is the case of sql type float(24) or real
-	if ( display_size == real_display_size ) {
-		os.precision( real_precision );
-	}
-	// this is the case of sql type float(53)
-	else if ( display_size == float_display_size ) {
-		os.precision( float_precision );
-	}
+    precision = get_float_precision( buffer_length, sizeof( Char ));
+    os.precision( precision );
     std::locale loc;
-    os.imbue( loc );
-    std::use_facet< std::num_put< Char > >( loc ).put( std::basic_ostream<Char>::_Iter( os.rdbuf() ), os, ' ', *number_data );
+    os.imbue(loc);
+    std::use_facet< std::num_put< Char > >( loc ).put( std::basic_ostream<Char>::_Iter( os.rdbuf()), os, ' ', *number_data );
     std::basic_string<Char>& str_num = os.str();
 
-    if( os.fail() ) {
-        last_error = new ( sqlsrv_malloc( sizeof( sqlsrv_error ))) sqlsrv_error( 
-            (SQLCHAR*) "IMSSP", (SQLCHAR*) "Failed to convert number to string", -1 );
+    if ( os.fail() ) {
+        last_error = new ( sqlsrv_malloc(sizeof( sqlsrv_error ))) sqlsrv_error(( SQLCHAR* ) "IMSSP", (SQLCHAR*) "Failed to convert number to string", -1 );
         return SQL_ERROR;
     }
 
-    if( str_num.size() * sizeof(Char) > (size_t) buffer_length ) {
-        last_error = new ( sqlsrv_malloc( sizeof( sqlsrv_error ))) sqlsrv_error( 
-            (SQLCHAR*) "HY090", (SQLCHAR*) "Buffer length too small to hold number as string", -1 );
+    if ( str_num.size() * sizeof( Char ) > ( size_t )buffer_length ) {
+        last_error = new ( sqlsrv_malloc(sizeof( sqlsrv_error ))) sqlsrv_error(( SQLCHAR* ) "HY090", ( SQLCHAR* ) "Buffer length too small to hold number as string", -1 );
         return SQL_ERROR;
     }
 
-    *out_buffer_length = str_num.size() * sizeof( Char ); // str_num.size() already include the NULL terminator
+   *out_buffer_length = str_num.size() * sizeof( Char ); // str_num.size() already include the NULL terminator
     memcpy_s( buffer, buffer_length, str_num.c_str(), *out_buffer_length );
 
     return SQL_SUCCESS;
+#else
+    std::basic_string<char> str_num;
+    SQLRETURN r;
```

Comment:
Just curious, have you confirmed that the code for string conversions written for Linux doesn't work in Windows?